### PR TITLE
Add authentication policy to bookmark app

### LINKED ISSUE #1352: Bookmark Application Sign-On Policy
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `okta_app_bookmark` resource does not have an input for an Authentication Policy ID. This is requiring Okta to request a second factor of authentication for our users when accessing from the Dashboard before the user is redirected to the bookmarked app.

The Okta Admin Portal **DOES** allow for configuration of Authentication Policies for Bookmarked apps.

### New or Affected Resource(s)

- okta_app_bookmark

### Potential Terraform Configuration

```hcl

data "okta_policy" "custom_policy" {
  name = "Custom Configured Policy"
  type = "ACCESS_POLICY"
}

resource "okta_app_bookmark" "example_app" {
  label                 = "Example App"
  url                   = "${var.example_app_url}"
  skip_groups = true
  # Desired:
  authentication_policy = okta_policy.custom_policy.id
}
```

### References
App Bookmark Resource Docs:
https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_bookmark

I'm unable to post a screenshot from the Okta UI, but it does allow for configuration of auth policy

Okta API Docs (no mention of bookmark application policy)
https://developer.okta.com/docs/reference/api/apps/#update-application-policy

### PR DESCRIPTION
**Pull request by jakezarobsky-8451**
closes #1352

### CODE REVIEW

**Comment by monde** [comment]
Thanks @jakezarobsky-8451 we need integration tests on PRs. I try to help out with this when I can. My bandwidth is low right now, if you could take a try at it the test is in https://github.com/okta/terraform-provider-okta/blob/master/okta/resource_okta_app_bookmark_test.go . For new refinements I like to inline the config like the example [TestAccAppBookmarkApplication_timeouts](https://github.com/okta/terraform-provider-okta/blob/master/okta/resource_okta_app_bookmark_test.go#L55-L64)

**Comment by jakezarobsky-8451** [comment]
I'll try it out - was going to add a test but didn't see a good example for the authentication policy in the other apps - which is why I didn't include it originally.

**Comment by jakezarobsky-8451** [comment]
@monde  - I think I updated this successfully - didn't end up inlining the configs - just to minimize the changes in the PR. If that is really necessary for merging I can do it - do you want me to do anything with the examples if I do make that change?

**Comment by monde** [comment]
@jakezarobsky-8451 this looks good. Since this is net-new, that is good to add it in the examples files. You reminded me, when it's a bug being fixed I prefer to inline the configs in the ITs and not clutter up clean examples. This also shows the config next to test demonstrating the fix has occurred.

I'll run this locally later today and finish the review.

**Comment by jakezarobsky-8451** [comment]
@monde  - Sorry to be a thorn in the side - were you able to see if this would run locally?

**Comment by monde** [comment]
🤔 you didn't actually run your tests:
```
for t in `grep "func Test" okta/resource_okta_app_bookmark_test.go | awk '{ print $2}'| sed -e 's/(.*//g'`; do
TF_LOG=warn TF_ACC=1 go test -tags unit -mod=readonly -test.v -run ^${t}$ ./okta
done
# github.com/okta/terraform-provider-okta/okta [github.com/okta/terraform-provider-okta/okta.test]
okta/resource_okta_app_bookmark.go:39:5: syntax error: unexpected newline in composite literal; possibly missing comma or }
FAIL    github.com/okta/terraform-provider-okta/okta [build failed]
FAIL
# github.com/okta/terraform-provider-okta/okta [github.com/okta/terraform-provider-okta/okta.test]
okta/resource_okta_app_bookmark.go:39:5: syntax error: unexpected newline in composite literal; possibly missing comma or }
FAIL    github.com/okta/terraform-provider-okta/okta [build failed]
FAIL
```

once you fix your code I need you to also run and make any recommended fixes
```
make vet
```

File: examples/okta_app_bookmark/basic.tf [hunk 1]
```diff
@@ -3,7 +3,8 @@ resource "okta_group" "group" {
 }
 
 resource "okta_app_bookmark" "test" {
-  label  = "testAcc_replace_with_uuid"
-  url    = "https://test.com"
-  groups = [okta_group.group.id]
+  label                 = "testAcc_replace_with_uuid"
+  url                   = "https://test.com"
+  authentication_policy = "some-authentication-policy-id"
+  groups                 = [okta_group.group.id]
```
Comment (monde) [inline examples/okta_app_bookmark/basic.tf]: nit, lining up whitespace and equals

File: okta/resource_okta_app_bookmark.go [hunk 2]
```diff
@@ -32,6 +32,11 @@ func resourceAppBookmark() *schema.Resource {
 				Default:  false,
 				Optional: true,
 			},
+			"authentication_policy": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Id of this apps authentication policy",
+			}
```
Comment (monde) [inline okta/resource_okta_app_bookmark.go]:
```
okta/resource_okta_app_bookmark.go:39:5: syntax error: unexpected newline in composite literal; possibly missing comma or }
```

**Review by monde — CHANGES_REQUESTED** [review]

**Comment by jakezarobsky-8451** [comment]
That's definitely my bad - never written go before and thought I got the syntax down. I've installed the binary, run the formatter, corrected the tests. 

I'm thinking the ACC tests are still going to fail just because there's not a data reference to a specific authentication policy in the ACC test tenant - I didn't see authentication_policy on other tests

**Comment by monde** [comment]
Thanks @jakezarobsky-8451 , I won't haze you anymore 😺 
I'll start a local branch, merge in your changes and fix the tests. Thanks for your contribution.

**Comment by jakezarobsky-8451** [comment]
Thank you Mike!

**Pull request merged**